### PR TITLE
Collect CREngine child

### PR DESCRIPTION
**Pull request by AntonKozlov**
When VM calls for CREngine it does not properly awaits termination of the engine. In case of `criuengine`, this could lead for a dangling process in the CRIU image [1]. This patch adds missing waitpid that eliminates the problem.

[1] https://github.com/openjdk/crac/blob/crac/src/java.base/unix/native/criuengine/criuengine.c#L68

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Reviewers
 * [Dan Heidinga](https://openjdk.java.net/census#heidinga) (@DanHeidinga - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/crac pull/18/head:pull/18` \
`$ git checkout pull/18`

Update a local copy of the PR: \
`$ git checkout pull/18` \
`$ git pull https://git.openjdk.java.net/crac pull/18/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18`

View PR using the GUI difftool: \
`$ git pr show -t 18`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/crac/pull/18.diff">https://git.openjdk.java.net/crac/pull/18.diff</a>

</details>

### CODE REVIEW

**Comment by AntonKozlov** [comment]
Ping? This a bit annoying problem that we checkpoint/restore an unnecessary process.

File: src/hotspot/os/linux/os_linux.cpp [hunk 1]
```diff
@@ -5826,13 +5826,27 @@ static bool compute_crengine() {
 }
 
 static int call_crengine() {
-  if (_crengine && !fork()) {
+  if (!_crengine) {
+    return -1;
+  }
+
+  pid_t pid = fork();
```
Comment (DanHeidinga) [inline src/hotspot/os/linux/os_linux.cpp:5833]: Should we check for fork() to fail?  If so, we need to handle `-1` separately from `0` (child) and `> 0` (parent)

**Review by DanHeidinga — COMMENTED** [review]
Apart from the possible fork failure case question, this looks reasonable

**Comment by DanHeidinga** [comment]
And not really related to this PR, but didn't we disable the non linux x64 pr builds?  Do we need to relook at that change?

**Comment by AntonKozlov** [comment]
I've added the check for the failed fork, thanks! I forgot to merge upstream changes to the branch, now the status of the PR is green.

**Review by DanHeidinga — APPROVED** [review]
Looks good to me

**Comment by AntonKozlov** [comment]
Thanks for review!

/integrate

**Pull request closed without merging**